python/ray/serve/_private/proxy_state.py

```diff
@@ -217,18 +235,24 @@ def update(self, draining: bool = False):
         ):
             return
 
+        # Doing a linear backoff for the ready check timeout.
+        ready_check_timeout = (
+            self.proxy_restart_count + 1
+        ) * PROXY_READY_CHECK_TIMEOUT_S
         if self._status == ProxyStatus.STARTING:
-            finished, _ = ray.wait([self._ready_obj_ref], timeout=0)
+            finished, _ = self._ray_api_wrapper.wait([self._ready_obj_ref], timeout=0)
```

Comment:
Similarly here we can have an `is_ready` method instead of directly calling the Ray API.